File: samples/basic-samples/simple-stack-example-basic-java-fragment/src/main/java/com/zhuinden/navigationexamplefrag/core/navigation/BaseFragment.java
package com.zhuinden.navigationexamplefrag.core.navigation;

import androidx.fragment.app.Fragment;

/**
 * Created by Owner on 2017. 06. 29..
 */

public class BaseFragment extends Fragment {
    public final <T extends BaseKey> T getKey() {
        return getArguments() != null ? getArguments().<T>getParcelable("KEY") : null;
    }
}

File: examples/basic-examples/simple-stack-example-basic-kotlin-fragment/src/main/java/com/zhuinden/simplestackexamplekotlinfragment/core/navigation/BaseFragment.kt
package com.zhuinden.simplestackexamplekotlinfragment.core.navigation

import androidx.fragment.app.Fragment
import com.zhuinden.simplestackexamplekotlinfragment.utils.requireArguments

/**
 * Created by Owner on 2017.11.13.
 */

open class BaseFragment : Fragment() {
    fun <T : BaseKey> getKey(): T = requireArguments.getParcelable<T>("KEY")!!
}
